`global` is not defined in the browser

The check for if the `TextDecoder` polyfill is required currently references `global`, which only exists on Node.js and not in the browser.

https://github.com/near/borsh-js/blob/261d9cde21665de5a9e2c75ec2509c0bfb7030bf/borsh-ts/index.ts#L6-L9

With bundlers that don't automatically polyfill `global` this causes problems and requires hacks to make it work. What you would think of one of the following solutions?

- `typeof TextDecoder !== "function" ? encoding.TextDecoder : TextDecoder`
- `typeof globalThis.TextDecoder !== "function" ? encoding.TextDecoder : globalThis.TextDecoder`

`globalThis` has been available for a while now, but I think the first one is best since it doesn't make any assumptions about the name of the global variable. Any thoughts? I'm happy to submit a PR for this.

Shouldn't the TypeScript compiler check the availability of the `globalThis` variable?

```ts
// cat index.ts 
console.log(global, globalThis, global == globalThis);
```

```bash
$ yarn run tsc index.ts
index.ts:1:13 - error TS2304: Cannot find name 'global'.

1 console.log(global, globalThis, global == globalThis);
              ~~~~~~

index.ts:1:33 - error TS2304: Cannot find name 'global'.

1 console.log(global, globalThis, global == globalThis);
                                  ~~~~~~


Found 2 errors.
```

--- a/borsh-ts/index.ts
+++ b/borsh-ts/index.ts
@@ -3,11 +3,11 @@ import bs58 from "bs58";
 
 // TODO: Make sure this polyfill not included when not required
 import * as encoding from "text-encoding-utf-8";
-const TextDecoder =
-  typeof (global as any).TextDecoder !== "function"
+const ResolvedTextDecoder =
+  typeof TextDecoder !== "function"
     ? encoding.TextDecoder
-    : (global as any).TextDecoder;
-const textDecoder = new TextDecoder("utf-8", { fatal: true });
+    : TextDecoder;
+const textDecoder = new ResolvedTextDecoder("utf-8", { fatal: true });
 
 export function baseEncode(value: Uint8Array | string): string {
   if (typeof value === "string") {

--- a/lib/index.js
+++ b/lib/index.js
@@ -33,10 +33,10 @@ const bn_js_1 = __importDefault(require("bn.js"));
 const bs58_1 = __importDefault(require("bs58"));
 // TODO: Make sure this polyfill not included when not required
 const encoding = __importStar(require("text-encoding-utf-8"));
-const TextDecoder = typeof global.TextDecoder !== "function"
+const ResolvedTextDecoder = typeof TextDecoder !== "function"
     ? encoding.TextDecoder
-    : global.TextDecoder;
-const textDecoder = new TextDecoder("utf-8", { fatal: true });
+    : TextDecoder;
+const textDecoder = new ResolvedTextDecoder("utf-8", { fatal: true });
 function baseEncode(value) {
     if (typeof value === "string") {
         value = Buffer.from(value, "utf8");

